dotnet/samples/Concepts/Search/Google_TextSearch.cs

```diff
@@ -0,0 +1,172 @@
+        await foreach (Google.Apis.CustomSearchAPI.v1.Data.Result result in fullResults.Results)
+        {
+            Console.WriteLine($"Title:       {result.Title}");
+            Console.WriteLine($"Snippet:     {result.Snippet}");
+            Console.WriteLine($"Link:        {result.Link}");
+            Console.WriteLine($"DisplayLink: {result.DisplayLink}");
+            Console.WriteLine($"Kind:        {result.Kind}");
+            Console.WriteLine(new string('—', HorizontalRuleLength));
+        }
+    }
+
+    /// <summary>
+    /// Show how to create a <see cref="GoogleTextSearch"/> with a custom mapper and use it to perform a text search.
+    /// </summary>
+    [Fact]
+    public async Task UsingGoogleTextSearchWithACustomMapperAsync()
+    {
+        // Create an ITextSearch instance using Google search
+        var textSearch = new GoogleTextSearch(
+            searchEngineId: TestConfiguration.Google.SearchEngineId,
+            apiKey: TestConfiguration.Google.ApiKey,
+            options: new() { MapToString = webPage => JsonSerializer.Serialize(webPage) });
+
+        var query = "What is the Semantic Kernel?";
+
+        // Search with TextSearchResult textResult type
+        KernelSearchResults<string> stringResults = await textSearch.SearchAsync(query, new() { Count = 2, Offset = 0 });
+        Console.WriteLine("--- Serialized JSON Results ---");
+        await foreach (string result in stringResults.Results)
+        {
+            Console.WriteLine(result);
+            Console.WriteLine(new string('-', HorizontalRuleLength));
+        }
+    }
+
+    /// <summary>
+    /// Show how to create a <see cref="GoogleTextSearch"/> with a custom mapper and use it to perform a text search.
+    /// </summary>
+    [Fact]
+    public async Task UsingGoogleTextSearchWithASiteSearchFilterAsync()
+    {
+        // Create an ITextSearch instance using Google search
+        var textSearch = new GoogleTextSearch(
+            initializer: new() { ApiKey = TestConfiguration.Google.ApiKey, HttpClientFactory = new CustomHttpClientFactory(this.Output) },
+            searchEngineId: TestConfiguration.Google.SearchEngineId);
+
+        var query = "What is the Semantic Kernel?";
+
+        // Search with TextSearchResult textResult type
+        TextSearchOptions searchOptions = new() { Count = 4, Offset = 0, BasicFilter = new BasicFilterOptions().Equality("siteSearch", "devblogs.microsoft.com") };
+        KernelSearchResults<TextSearchResult> textResults = await textSearch.GetTextSearchResultsAsync(query, searchOptions);
+        Console.WriteLine("--- Microsoft Developer Blogs Results ---");
+        await foreach (TextSearchResult result in textResults.Results)
+        {
+            Console.WriteLine(result.Link);
+            Console.WriteLine(new string('-', HorizontalRuleLength));
+        }
+    }
+
+    #region private
+    private const int HorizontalRuleLength = 200;
```

Comment:
Default size of the console window itself is 80 chars wide (counting the path) per 25 lines height.

Suggest keeping the width as minimum as possible to avoid wrapping of rule.




Windows 
```suggestion
    private const int HorizontalRuleLength = 10;
```